src/GrammarApplicator_runRules.cpp

```diff
@@ -422,6 +422,11 @@ bool GrammarApplicator::runSingleRule(SingleWindow& current, const Rule& rule, R
 			}
 		}
 
+		// If this is SETPARENT SAFE and there's already a parent, skip it.
+		if (type == K_SETPARENT && rule.flags & RF_SAFE && cohort->dep_parent != DEP_NO_PARENT) {
```

Comment:
I want `()` around chained `a && (b & c)` - I know it's not necessary here, but too many bugs in the wild about `=` with `&` and `|`.